src/librbd/WatchNotifyTypes.cc

```diff
@@ -208,16 +208,22 @@ void AsyncCompletePayload::dump(Formatter *f) const {
 
 void ResizePayload::encode(bufferlist &bl) const {
   ::encode(size, bl);
+  ::encode(allow_shrink, bl);
   AsyncRequestPayloadBase::encode(bl);
 }
 
 void ResizePayload::decode(__u8 version, bufferlist::iterator &iter) {
   ::decode(size, iter);
+
+  if (version >= 3) {
```

Comment:
Correction -- should be version 4 and update the encoded version here: https://github.com/neurodrone/ceph/blob/950db41a78dac0d07d25436e677d4e5e7b6adae8/src/librbd/WatchNotifyTypes.cc#L284